### PR TITLE
Use Nested Namespaces resembling directory structure

### PR DESCRIPTION
**Pull request by ajaybhat**
This is a fix for #2752 
I have started with the <code>shogun.distance</code> package.
There is a lot of refactoring to be done, however I'd like to know if this is the way to go.

@lisitsyn @karlnapf

### CODE REVIEW

**Comment by karlnapf** [comment]
@lisitsyn has to comment here. I am not sure what is the most elegant way to go ahead here

**Comment by lisitsyn** [comment]
Hmm I am so sorry I missed this PR. Gonna glance over it now

File: src/shogun/distance/AttenuatedEuclideanDistance.h [hunk 1]
```diff
@@ -17,7 +17,9 @@
 #include <shogun/distance/RealDistance.h>
 #include <shogun/features/DenseFeatures.h>
 
-namespace shogun
+using namespace shogun;
```
Comment (lisitsyn) [inline src/shogun/distance/AttenuatedEuclideanDistance.h]: Here and everywhere when we do ```using namespace``` and then declare namespace, we don't do it nested. I think it should be like 
```namespace shogun
{
  namespace distance
  {```

Comment (ajaybhat) [inline src/shogun/distance/AttenuatedEuclideanDistance.h]: @lisitsyn Is use of the scope resolution operator <code>::</code> for specifying type of the fields of the different classes okay or is it something that you discourage?

Comment (lisitsyn) [inline src/shogun/distance/AttenuatedEuclideanDistance.h]: I mean we better have namespace `shogun::distance` but currently it is just `distance`

**Comment by karlnapf** [comment]
Any updates here?

**Comment by karlnapf** [comment]
closing as inactive

**Pull request closed without merging**